reporting.txt

```diff
@@ -0,0 +1,952 @@
+   Section 3)
+
+       {
+            "organization-name": organization-name,
+            "date-range": {
+                "start-datetime": date-time,
+                "end-datetime": date-time
+            },
+            "contact-info": email-address,
+            "report-id": report-id,
+            "applied-policy": {
+                "policy-type": policy-type,
+                "policy-string": policy-string,
+                "policy-domain": domain,
+                "mx-host": mx-host-pattern,
+                "policy-id": policy-id
+            },
+            "report-items": [
+               {
+                   "result-type": result-type,
+                   "sending-mta-ip": ip-address,
+                   "receiving-mta-hostname": receiving-mta-hostname,
+                   "message-count": message-count,
+                   "additional-information": additional-info-uri
+               }
+            ]
+       }
+
+   Figure x: JSON Report Format
+
+   o  "organization-name": The name of the organization responsible for
+      the report.  It is provided as a string.
+
+   o  "date-time": The date-time indicates the start- and end-times for
+      the report range.  It is provided as a string formatted according
+      to Section 5.6, "Internet Date/Time Format", of [RFC3339].
+
+   o  "email-address": The contact information for a responsible party
+      of the report.  It is provided as a string formatted according to
+      Section 3.4.1, "Addr-Spec", of [RFC5322].
+
+   o  "report-id": A unique identifier for the report.  Report authors
+      may use whatever scheme they prefer to generate a unique
+      identifier.  It is provided as a string.
+
+
+
+
+Margolis, et al.        Expires October 20, 2016               [Page 13]
+
+Internet-Draft                 SMTP-TLSRPT                    April 2016
+
+
+   o  "policy-type": The type of policy that was applied by the sending
+      domain.  Presently, the only two valid choices are "tlsa" and
+      "sts".  It is provided as a string.
+
+   o  "policy-string": The string serialization of the policy, whether
+      TLSA record or STS policy.  Any linefeeds from the original policy
+      MUST be replaced with [SP].  TODO: Help with specifics.
+
+   o  "domain": The Policy Domain upon which the policy was applied.
+      For messages sent to "user@example.com" this field would contain
+      "example.com".  It is provided as a string.
+
+   o  "mx-host-pattern": The pattern of MX hostnames from the applied
+      policy.  It is provided as a string.
+
+   o  "policy-id": The version number of the STS policy, or left blank
+      for TLSA.
+
+   o  "result-type": A value from the _Result Types_ section above.
+
+   o  "ip-address": The IP address of the sending MTA that attempted the
+      STARTTLS connection.  It is provided as a string representation of
+      an IPv4 or IPv6 address.
+
+   o  "receiving-mta-hostname": The hostname of the receiving MTA with
```

Comment:
Is this meant to be the HELO that was from the attempt or the DNS name that the IP resolves to?